src/Connector.Common/Clients/ConnectionAndTransaction.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Data;
+using System.Threading.Tasks;
+
+namespace CluedIn.Connector.Common.Clients
+{
+    public class ConnectionAndTransaction<TConnection, TTransaction> : IAsyncDisposable
+        where TConnection : IDbConnection
+        where TTransaction : IDbTransaction
+    {
+        public readonly TConnection Connection;
+        public readonly TTransaction Transaction;
+
+        public ConnectionAndTransaction(TConnection connection, TTransaction transaction)
+        {
+            Connection = connection;
+            this.Transaction = transaction;
+        }
+
+
+        public ValueTask DisposeAsync()
+        {
+            return new ValueTask(new Task(() =>
+            {
+                Transaction.Dispose();
```

Comment:
We are we not calling AsyncDispose here?

Reply:
We're not using `AsyncDispose`, since `IDbConnection` and `ITransaction` is not `AsyncDispose` by them. It would require adding that as a constraint a lot of places. I can add that if you think it's worth it

Reply:
Yeah ok the interface themselves isn't.

```
  public abstract class DbTransaction : 
    MarshalByRefObject,
    IDbTransaction,
    IDisposable,
    IAsyncDisposable
```

What about?:

```
if (Transaction is IAsyncDisposable t)
   await t.DisposeAsync();
else
   Transaction.Dispose();
```

Reply:
Done